crates/ef-testing/src/sequencer/utils.rs

```diff
@@ -0,0 +1,92 @@
+use blockifier::abi::abi_utils::get_storage_var_address as blockifier_get_storage_var_address;
+use reth_primitives::{Bytes, TransactionSigned};
+use reth_rlp::Decodable;
+use revm_primitives::U256;
+use starknet::{
+    core::types::{BroadcastedInvokeTransaction, FieldElement},
+    macros::selector,
+};
+use starknet_api::{
+    core::{ClassHash, ContractAddress},
+    hash::StarkFelt,
+    state::StorageKey,
+};
+
+use super::{constants::KAKAROT_ADDRESS, KakarotSequencer};
+
+pub(crate) fn get_storage_var_address(
+    storage_var: &str,
+    keys: &[StarkFelt],
+) -> Result<StorageKey, eyre::Error> {
+    Ok(blockifier_get_storage_var_address(storage_var, keys)?)
+}
+
+pub(crate) fn split_bytecode_to_starkfelt(bytecode: &Bytes) -> Vec<StarkFelt> {
+    bytecode
+        .chunks(16)
+        .map(|x| {
+            let mut storage_value = [0u8; 16];
+            storage_value[..x.len()].copy_from_slice(x);
+            StarkFelt::from(u128::from_be_bytes(storage_value))
+        })
+        .collect()
+}
+
+pub(crate) fn split_u256(value: U256) -> [u128; 2] {
+    [
+        (value & U256::from(u128::MAX)).try_into().unwrap(), // safe unwrap <= U128::MAX.
+        (value >> 128).try_into().unwrap(),                  // safe unwrap <= U128::MAX.
+    ]
+}
+
+pub(crate) fn contract_address_to_starkfelt(contract_address: &ContractAddress) -> StarkFelt {
+    *contract_address.0.key()
+}
+
+pub(crate) fn class_hash_to_starkfelt(class_hash: &ClassHash) -> StarkFelt {
+    class_hash.0
+}
+
+pub fn bytes_to_felt_vec(bytes: &Bytes) -> Vec<FieldElement> {
+    bytes.to_vec().into_iter().map(FieldElement::from).collect()
+}
+
+#[allow(dead_code)]
+pub(crate) fn to_broadcasted_starknet_transaction(
```

Comment:
what is this used for? also, even though rust compiler resolves circular deps, I'm always quite worried about their introduction as it often is a sign a "feature envy"

Reply:
and actuallt ("suprisingly") the `Kakarot.compute_starknet_address` doesn't even use `self`

Reply:
removed the self which wasn't needed indeed. I am not sure what you mean by circular dep in this case. This function is used to convert a rlp encoded evm transaction into a starknet transaction.

Reply:
`sequencer/utils.rs` imports `use super::{constants::KAKAROT_ADDRESS, KakarotSequencer};`

and `mod.rs` imports

```
use self::utils::{
    class_hash_to_starkfelt, contract_address_to_starkfelt, get_storage_var_address,
};

pub(crate) struct KakarotSequencer(Sequencer<State>);
```

so A imports B imports A, it's circular deps and it wouldn't work in python for example.

It means here that the `compute_starknet_address` should probably not be in the `KakarotSequencer`

Reply:
will move it to utils